Add a switch to use redfish as BMC driver

kashifest opened the pull request:
Add a switch for using redfish bmc driver

kashifest committed:
Add a switch to use redfish as BMC

maelk commented on config_example.sh:
```diff
@@ -74,3 +74,8 @@
+#
+# Set the driver
+#
+#export BMC_DRIVER="redfish"
```
add a new line

maelk commented on vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml:
```diff
@@ -112,6 +112,24 @@
+        vm_id: "{{ vm_id|default({}) | combine ( {item.item.name: item.stdout} ) }}"
+      with_items: "{{ vm_uuid.results }}"
+    
+    - name: print variable
```
remove this, it is only for development

maelk commented on vm-setup/roles/libvirt/templates/ironic_nodes.json.j2:
```diff
@@ -10,15 +10,22 @@
-        "ipmi_port": "{{ node.virtualbmc_port }}",
+        "username": "admin",
+        "password": "password",
+        "ip": "{{ lvars['host_ip'] }}",
```
I don't think we need this field

maelk commented on vm-setup/roles/libvirt/templates/ironic_nodes.json.j2:
```diff
@@ -10,15 +10,22 @@
+        "address": "{{driver}}+{{"http://"}}{{ lvars['host_ip'] }}{{":8000/redfish/v1/Systems/"}}{{vm_id[node.name]}}",
+        {% else -%}
+        "port": "{{ node.virtualbmc_port }}",
+        "address": "{{driver}}{{"://"}}{{lvars['host_ip']}}{{":"}}{{node.virtualbmc_port}}",
```
this needs a condition, if the port is defined, otherwise there will be a formatting problem

maelk commented on vm-setup/roles/libvirt/templates/ironic_nodes.json.j2:
```diff
@@ -10,15 +10,22 @@
+        "ip": "{{ lvars['host_ip'] }}",
+        {% if driver =='redfish' -%}
+        "port": "8000",
+        "address": "{{driver}}+{{"http://"}}{{ lvars['host_ip'] }}{{":8000/redfish/v1/Systems/"}}{{vm_id[node.name]}}",
```
the {{ }} around strings are not needed, only around vars

kashifest replied:
I wanted to keep it as is to make it more readable

Commit:
Reinstate virtualbmc service teardown tasks

These were removed in #48 which is fine for a greenfield CI environment
but leads to unexpected behavior in developer test environments, since
the old systemd service is not stopped before starting the new containers.

russellb committed:
Merge pull request #63 from hardys/vbmc_fixes

Reinstate virtualbmc service teardown tasks

Commit:
source lib/common.sh in 01_prepare_host.sh

Currently VBMC_IMAGE and SUSHY_TOOLS_IMAGE are undefined because
they're defined in lib/common.sh.

Also I noticed that the pushd uses a relative path which won't work
unless the calling shell is in the metal3-dev-env dir, and we're
not logging the output from this script.

russellb committed:
Merge pull request #64 from hardys/vbmc_fixes_2

source lib/common.sh in 01_prepare_host.sh

russellb committed:
Use vbmc and sushy-tools images from quay.io.

Instead of building these locally, pull pre-built images from quay.io.
These images are set to automatically rebuild any time metal3-dev-env
changes.

russellb committed:
Merge pull request #65 from russellb/quay-images

Use vbmc and sushy-tools images from quay.io.

kashifest commented:
/test-integration

nordixinfra reviewed (commented):
:revolving_hearts: Ubuntu Test pending : https://jenkins.nordix.org/job/airship_nordix_metal3_dev_env_integration_test_ubuntu/5/

nordixinfra reviewed (commented):
:broken_heart: Test Failure Ubuntu : https://jenkins.nordix.org/job/airship_nordix_metal3_dev_env_integration_test_ubuntu/5/
